Userland/Libraries/LibGfx/Filters/StackBlurFilter.cpp

```diff
@@ -0,0 +1,329 @@
+    329, 326, 323, 320, 318, 315, 312, 310, 307, 304, 302, 299, 297, 294, 292,
+    289, 287, 285, 282, 280, 278, 275, 273, 271, 269, 267, 265, 263, 261, 259
+};
+
+constexpr Array<u8, 255> shift_table {
+    9, 11, 12, 13, 13, 14, 14, 15, 15, 15, 15, 16, 16, 16, 16, 17,
+    17, 17, 17, 17, 17, 17, 18, 18, 18, 18, 18, 18, 18, 18, 18, 19,
+    19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 20, 20, 20,
+    20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 21,
+    21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21,
+    21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 22, 22, 22, 22, 22, 22,
+    22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22,
+    22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 22, 23,
+    23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23,
+    23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23,
+    23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23,
+    23, 23, 23, 23, 23, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,
+    24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,
+    24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,
+    24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24,
+    24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24
+};
+
+// Note: This is named to be consistent with the algorithm, but it's actually a simple circular buffer.
+struct BlurStack {
+    BlurStack(size_t size)
+    {
+        m_data.resize(size);
+    }
+
+    struct Iterator {
+        friend BlurStack;
+
+        ALWAYS_INLINE Color& operator*()
+        {
+            return m_data.at(m_idx);
+        }
+
+        ALWAYS_INLINE Color* operator->()
+        {
+            return &m_data.at(m_idx);
+        }
+
+        ALWAYS_INLINE Iterator operator++()
+        {
+            // Note: This seemed to profile slightly better than %
+            if (++m_idx >= m_data.size())
+                m_idx = 0;
+            return *this;
+        }
+
+        ALWAYS_INLINE Iterator operator++(int)
+        {
+            auto prev_it = *this;
+            ++*(this);
+            return prev_it;
+        }
+
+    private:
+        Iterator(size_t idx, Span<Color> data)
+            : m_idx(idx)
+            , m_data(data)
+        {
+        }
+
+        size_t m_idx;
+        Span<Color> m_data;
+    };
+
+    Iterator iterator_from_position(size_t position)
+    {
+        VERIFY(position < m_data.size());
+        return Iterator(position, m_data);
+    }
+
+private:
+    Vector<Color, 512> m_data;
+};
+
+// This is an implementation of StackBlur by Mario Klingemann (https://observablehq.com/@jobleonard/mario-klingemans-stackblur)
+// (Link is to a secondary source as the original site is now down)
+FLATTEN void StackBlurFilter::process_rgba(u8 radius)
+{
+    // TODO: Implement a plain RGB version of this (if required)
+
+    using uint = unsigned;
```

Comment:
Can we use u32 in all cases instead?

Reply:
I kinda prefer using `uint` here since `u32` (to me anyway) suggests that it needs 32 bits. It actually only needs up to 17 bits here.

Reply:
well thats unfortunate, just off a `short`...
lets let a mainteainer make the call here

Reply:
A comment about the used size would be nice, if we know that though

Reply:
I think if you limit the max radius to 180px it fits in a `u16`. Maybe that's the thing to do since any blur above about 20px is super rare (and I'm not sure a really high radius above 180px even makes sense for a box-shadow).

Reply:
This just seems excessive, why not use `unsigned` instead of giving it a name that means exactly the same thing?

Reply:
Because `unsigned` is annoyingly long (and I always type it wrong)